docs/contributing.rst

```diff
@@ -62,26 +62,210 @@ The following Python implementations are supported:
+enabling and disabling the instrumentation and overriding the service name.
+
+
+Library Support
+~~~~~~~~~~~~~~~
+``ddtrace`` tries to support as many active versions of a library as possible.
+Because of this, it can become tricky to instrument a library due to changing
+features and APIs.
+
+For tricky libraries it's recommended to pull out the version of the library to
+use when instrumenting volatile features. A great example of this is the Flask
+integration:
+
+    - pulling out the version: `flask version <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/flask/patch.py#L45-L58>`_
+    - using it to instrument a later-added feature `flask version usage <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/flask/patch.py#L149-L151>`_
+
+Exceptions/Errors
+~~~~~~~~~~~~~~~~~
+
+Exceptions provide a lot of useful information about errors and the application
+as a whole and are fortunately usually quite easy to deal with. Exceptions are
+a great place to start instrumenting. There are a couple of considerations when
+dealing with exceptions in ``ddtrace``:
+
+    - Re-raising the exception: it is crucial that we do not interfere with the
+      application, so exceptions must be re-raised. See the `bottle exception handling <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/bottle/trace.py#L50-L69>`_
+      instrumentation for an example.
+
+    - Gathering relevant information: exceptions usually contain a lot of
+      relevant information for tracking down a bug. ``ddtrace`` provides
+      a helper for pulling out this information and adding it to a span.
+      See the `cassandra exception handling <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/cassandra/session.py#L117-L122>`_
+      instrumentation for an example.
+
+Distributed Tracing
+~~~~~~~~~~~~~~~~~~~
+
+Some integrations pass information across application boundaries to other
+applications where the request is continued. Datadog and ``ddtrace`` provide
+support for continuing a trace in another application. Distributed tracing only makes
+sense for libraries that send or receive requests across application boundaries.
+
+    - Propagating the trace example: `requests <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/requests/connection.py#L85-L88>`_
+    - Receiving a propagated trace example: `Django <https://github.com/DataDog/dd-trace-py/blob/96dc6403e329da87fe40a1e912ce72f2b452d65c/ddtrace/contrib/django/middleware.py#L116-L121>`_
+
+
+Web frameworks
+++++++++++++++
+Web frameworks centre around web requests and hence so does the tracing. A
+web framework integration must do the following if possible:
+
+    - Trace the duration of the request.
+    - Report HTTP status codes.
```

Comment:
Use set_http_meta() helper in trace_utils.py to set http attributes (method, url, status code, query params, request headers)